Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskSet.cs

```diff
@@ -331,11 +332,71 @@ public void ReleaseCancelRequestsWriter()
         {
             CancelRequestsDataStream.ReleasePending();
         }
+        
+        //*************************************************************************************************************
+        // MIGRATION
+        //*************************************************************************************************************
+
+        public void AddToMigrationLookup(
+            string parentPath, 
+            Dictionary<string, uint> migrationActiveIDLookup,
+            PersistentDataSystem persistentDataSystem)
+        {
+            foreach (KeyValuePair<Type, AbstractDataStream> entry in m_PublicDataStreamsByType)
+            {
+                AddToMigrationLookup(parentPath, entry.Key.GetReadableName(), entry.Value.ActiveID, migrationActiveIDLookup);
+            }
+
+            foreach (ICancellableDataStream entry in m_DataStreamsWithExplicitCancellation)
+            {
+                AddToMigrationLookup(parentPath, $"{entry.InstanceType.GetReadableName()}-ExplicitCancel", entry.PendingCancelActiveID, migrationActiveIDLookup);
+            }
+            
+            AddToMigrationLookup(
+                parentPath, 
+                typeof(CancelRequestsDataStream).GetReadableName(), 
+                CancelRequestsDataStream.ActiveID, 
+                migrationActiveIDLookup);
+            
+            AddToMigrationLookup(
+                parentPath, 
+                typeof(CancelProgressDataStream).GetReadableName(), 
+                CancelProgressDataStream.ActiveID, 
+                migrationActiveIDLookup);
+            
+            AddToMigrationLookup(
+                parentPath, 
+                typeof(CancelCompleteDataStream).GetReadableName(), 
```

Comment:
Mild concern about using readable name for a uniqueID type situation.
Readable name is used for human readable formatting and if we ever changed that formatting it could break the path. Should we just use C#'s built in?

Reply:
Hmmmn I'm not sure. If we changed the formatting it would break, but only if we changed the formatting mid-run. It's a runtime only id. 

We could use the built in, it's just harder to parse when debugging. Thoughts?

Reply:
Hmmmn ok, fair point. I changed it to `FullName` for now to keep the uniqueness. 

I added a TODO to hash the `FullName` to get shorter paths.

Reply:
teeechhnically you want the Assembly qualified one

Reply:
Switched to using BurstRuntime to get the hash of the type which uses the Assembly qualified name under the hood